src/interface/update.hpp

```diff
@@ -34,48 +34,50 @@ namespace parthenon {
 namespace Update {
 
 template <typename T>
-TaskStatus FluxDivergence(std::shared_ptr<T> &in, std::shared_ptr<T> &dudt_obj);
+TaskStatus FluxDivergence(T *in, T *dudt_obj);
 
 template <typename F, typename T>
-void UpdateData(const std::vector<F> &flags, T &in, T &dudt, const Real dt, T &out) {
-  const auto &in_pack = in->PackVariables(flags);
-  auto out_pack = out->PackVariables(flags);
-  const auto &dudt_pack = dudt->PackVariables(flags);
+TaskStatus SumData(const std::vector<F> &flags, T *in1, T *in2, const Real w1,
```

Comment:
I like this. This is more general and more useful.

Nitpick on the name: This isn't really a sum, it's more like a fuse-multiply-add. How about `TaskStatus FMAData` or something similar?

Also should `w1` and `w2` default to `1.0`?

Reply:
It's not really a FMA either, is it?  How about `TaskStatus WeightedSumData`?  I think I like the idea to default the weights to 1...

Reply:
@Yurlungur pushed a change here.   Look OK?

Reply:
:+1: Yep I like it.